packages/adblocker/src/engine/engine.ts

```diff
@@ -834,6 +921,36 @@ export default class FilterEngine extends EventEmitter<
       isFilterExcluded: this.isFilterExcluded.bind(this),
     });
 
+    const context: CosmeticFilterMatchingContext = {
+      url,
+      domain,
+      hostname,
+
+      classes,
+      hrefs,
+      ids,
+
+      allowGenericHides,
+      allowSpecificHides,
+
+      getBaseRules,
+      getInjectionRules,
+      getExtendedRules,
+      getRulesFromDOM,
+      getRulesFromHostname,
+
+      matchType: FilterType.COSMETIC,
+      reference,
+    };
+
+    for (const match of matches) {
+      this.emit('filter-matched', match, context);
+
+      if (exceptionMatches.has(match)) {
+        this.emit('filter-matched', exceptionMatches.get(match)!, context);
+      }
+    }
```

Comment:
Can we do this only if there is at least one event listener register for `filter-matched`? Also, would it make sense to have a different event name for cosmetics and network?